### DISCUSSION TITLE
Removing of player dependencies from "@project-sunbird/sunbird-collection-editor"

### DISCUSSION THREAD

**Discussion by sedin-tushar** [Sunbird-Knowlg/Community#185] [Feature Requests]
**Description:**
We are integrating the Players as a web-components in the portal following the [Issue] (https://project-sunbird.atlassian.net/browse/ED-1473) and facing a dependency issue with sunbird-collection-editor.

**Steps Taken:**
As a part of the implementation we removed the `"@project-sunbird/sunbird-pdf-player-v9", @project-sunbird/sunbird-video-player-v9 and "@project-sunbird/sunbird-epub-player-v9"` from `package.json.`  

**Error Encountered:**
But since these Players have a dependency to the collection-editor libray we are getting the below error during the build
![Pdf](https://github.com/Sunbird-Knowlg/Community/assets/127084282/9555dea7-d192-4489-9b17-d89338cb0702)
![Epub](https://github.com/Sunbird-Knowlg/Community/assets/127084282/05a85e62-895e-4246-9425-fb4dd7b5810b)

**Request for Assistance**
Please remove these dependencies from the mentioned package. This will enable us to implement Players as web components successfully.

CC: @vinukumar-vs @rajeevsathish @princegupta1131

**Comment by HarishGangula** [comment]
@sedin-tushar  Can you share the version of the sunbird-collection-editor you are trying to integrate for better analysis.

**Reply by sedin-tushar** [reply]
@HarishGangula We are using `@project-sunbird/sunbird-collection-editor": "5.4.9` for integrating Players as web-components

**Comment by princegupta1131** [comment]
Hi @HarishGangula as we mentioned above, we are facing the error while removing the player package from the portal
```
Failed to initialize Angular compilation - The target entry-point "@project-sunbird/sunbird-collection-editor" has missing dependencies:

@project-sunbird/sunbird-pdf-player-v9
@project-sunbird/sunbird-epub-player-v9
@project-sunbird/sunbird-video-player-v9
@project-sunbird/sunbird-quml-player"
@project-sunbird/sunbird-quml-player-v9
```
It appears that the above packages are used as peer dependencies in the library, and their removal has caused these compilation errors.
Could you please provide the latest collection-editor version with this fix?
CC: @rajeevsathish @vinukumar-vs @mohitga @pallakartheekreddy

**Reply by HarishGangula** [reply]
@princegupta1131  Can we try https://www.npmjs.com/package/@project-sunbird/sunbird-collection-editor-web-component for integration as web component

**Reply by princegupta1131** [reply]
Sure @HarishGangula,
Let me try using the collection editor as a web component in the portal

**Reply by princegupta1131** [reply]
Hi, @HarishGangula I have integrated it as a web component. I have encountered a runtime error in the console, as we are using 
@project-sunbird/ckeditor-build-classic": "4.1.3" in the portal for dashboard feature, sb-forms and question-set-editor. It is conflicting with the CKEditor which is bundled in the web component.


**ERROR**
```
Uncaught CKEditorError: ckeditor-duplicated-modules
Read more: https://ckeditor.com/docs/ckeditor5/latest/framework/guides/support/error-codes.html#error-ckeditor-duplicated-modules
    at ckeditor.js:5:133019
    at ckeditor.js:5:816369
    at ckeditor.js:5:816385
    at ckeditor.js:5:18786
    at 25662 (ckeditor.js:5:18912)
    at __webpack_require__ (bootstrap:19:1)
 ```
 
 Please suggest on this issue

**Reply by HarishGangula** [reply]
This come if you have already ckEditor as dependecy that you can remove it

**Reply by princegupta1131** [reply]
HI @HarishGangula 
As per the discussion with @vinukumar-vs  28/03/2024, we've encountered the error "**Uncaught CKEditorError: ckeditor-duplicated-modules"**. It has been identified that this error arises due to CKEditor dependencies being initialized in both the web component and the parent application, resulting in duplicate module errors. To resolve this issue and prevent duplicate module initialization, it has been decided to address the CKEditor dependencies on the web component side. A new story has been created to address this in the upcoming planning session. 
[ED-3926: CKEditorError: ckeditor-duplicated-modules](https://project-sunbird.atlassian.net/browse/ED-3926)

CC:   @rajeevsathish  [rajnishdargan](https://github.com/rajnishdargan)
